client/db_test.go

```diff
@@ -0,0 +1,75 @@
+package client_test
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/server"
+)
+
+func ExampleDB() {
+	s := server.StartTestServer(nil)
+	defer s.Stop()
+
+	db := client.Open("https://root@" + s.ServingAddr())
+
+	if err := db.Put("key", "value"); err != nil {
+		log.Fatal(err)
+	}
+	result := db.Get("key")
+	fmt.Println(1, result.Rows[0].ValueStringer())
+
+	if r := db.Inc("inc", 100); r.Err != nil {
+		log.Fatal(r.Err)
+	}
+	result = db.Get("key", "inc")
+	fmt.Println(2, result.Rows[0].ValueStringer())
+	fmt.Println(3, result.Rows[1].ValueInt())
+
+	b := &client.Batch{}
+	b.Get("key")
+	b.Scan("i", "j", 100)
+	b.Put("foo", "bar")
+	if err := db.Run(b); err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println(4, b.Results[0].Rows[0].ValueStringer())
```

Comment:
How would the lookup by key be implemented? Scanning over the results? Seems bad to provide an interface that can't be implemented efficiently when the user can easily keep track of which row will have the correct key.

I'm not sure about the other suggestion regarding accessors for the results. It isn't called out in the non-existent documentation, but for `Get`, `Put`, `Inc` and `Del` there is well defined number of rows that will be returned. The docs should eventually say something like: Each operation (`Get`, `Put`, `Inc`, `Scan` or `Del`) will add a new `Result` to the batch. For `Get` and `Del`, the `Result` will contain exactly as many rows as the number of arguments passed to `Get` and `Del`. For `Put` and `Inc`, the `Result` will contain exactly 1 row. For `Scan` the number of rows is variable but is capped at the supplied `maxRows` parameter.